### DISCUSSION TITLE
withMeta().example() not working with input that come from router with route path params

### DISCUSSION THREAD

**Discussion by huyhoang160593** [RobinTail/express-zod-api#1365] [Q&A] [answered]
Hi @RobinTail, this maybe some more problem with the auto generate document, when I want to use them with the endpoint which include dynamic routing that come from params and the body is json(POST/PUT request), the wrap withMeta().example() currently not working with input at all, they only show the default version of example in all case. Can you suggest temp fix or a patch to fix this ???

**Comment by huyhoang160593** [comment]
> Hi @RobinTail, this maybe some more problem with the auto generate document, when I want to use them with the endpoint which include dynamic routing that come from params and the body is json(POST/PUT request), the wrap withMeta().example() currently not working with input at all, they only show the default version of example in all case. Can you suggest temp fix or a patch to fix this ???

Current version: "express-zod-api": "^14.1.0"
NodeJS version: 18.17.0

**Comment by RobinTail** [comment]
Hello @huyhoang160593 .
Thank you for informing me.
> Can you suggest temp fix or a patch to fix this ???

In order to do that, I need to understand the problem better.
Could you make a minimal code sample for reproducing the problem?
And when you say that `withMeta().example()` not working, do you mean that example does not reach the generated documentation when doing `new Documentation().getSpecAsYaml()` ? Or do you mean something else?
Please elaborate on sharing more information, so I could make a test, then find and fix the cause of the problem.

**Comment by RobinTail** [comment]
I have an endpoint in example that is using route path parameter and using that parameter in `withMeta().example()`.

https://github.com/RobinTail/express-zod-api/blob/99a282001c8a5cf8093d06f63165e75da6a86b0b/example/routing.ts#L17-L19

https://github.com/RobinTail/express-zod-api/blob/99a282001c8a5cf8093d06f63165e75da6a86b0b/example/endpoints/update-user.ts#L24-L25

Then it goes into the generated documentation:

https://github.com/RobinTail/express-zod-api/blob/99a282001c8a5cf8093d06f63165e75da6a86b0b/example/example.swagger.yaml#L169-L174

So, basically this case (at least how I understand it), is covered by tests and operational.
There is probably something else, or I don't understand some details.
That's why I need a code sample that fails, @huyhoang160593

**Reply by huyhoang160593** [reply, marked as answer]
@RobinTail yeah so after doing the minimal repoduction, I realise that it's totally my fail, cause of my complicated schema, so that the example is valid in the typescript but fail in zod checking, in that case the example will fail back to the default one @@. Thanks you for taking the time to provide more detail, I'm very appreciate that
